src/main/java/uk/gov/justice/hmpps/prison/service/BedAssignmentHistoryService.java

```diff
@@ -51,22 +52,37 @@ public void add(final Long bookingId, final Long livingUnitId, final String reas
     @VerifyBookingAccess
     public Page<BedAssignment> getBedAssignmentsHistory(final Long bookingId, final PageRequest pageRequest) {
         final var bedAssignmentsHistory = repository.findAllByBedAssignmentHistoryPKOffenderBookingId(bookingId, pageRequest);
-        final var results = bedAssignmentsHistory.getContent().stream().map(assignment -> {
-                    final var agencyInternalLocation = locationRepository.findOneByLocationId(assignment.getLivingUnitId());
-                    return BedAssignment.builder()
-                                    .livingUnitId(assignment.getLivingUnitId())
-                                    .description(agencyInternalLocation.map(AgencyInternalLocation::getDescription).orElse(null))
-                                    .assignmentDate(assignment.getAssignmentDate())
-                                    .assignmentEndDate(assignment.getAssignmentEndDate())
-                                    .assignmentDateTime(assignment.getAssignmentDateTime())
-                                    .assignmentEndDateTime(assignment.getAssignmentEndDateTime())
-                                    .assignmentReason(assignment.getAssignmentReason())
-                                    .bookingId(assignment.getOffenderBooking().getBookingId())
-                                    .agencyId(agencyInternalLocation.map(AgencyInternalLocation::getAgencyId).orElse(null))
-                                    .build();
-            })
-                        .collect(Collectors.toList());
+        final var results = bedAssignmentsHistory.getContent()
+                .stream()
+                .map(this::transform)
+                .collect(Collectors.toList());
+
         return new PageImpl<>(results, pageRequest, bedAssignmentsHistory.getTotalElements());
     }
 
+    public List<BedAssignment> getBedAssignmentsHistory(final long livingUnitId, final LocalDate from, final LocalDate to) {
+        if (!locationRepository.existsById(livingUnitId)) throw new EntityNotFoundException(String.format("Cell %s not found", livingUnitId));
```

Comment:
Perhaps check the dates are valid/correct way round?

Reply:
Not sure we need to check the format as it's dealt with at the controller level. However, I agree that the date order should be validated.

Reply:
Not sure on the logic here :

```
if (from.isAfter(to) && !from.isEqual(to)) {
  fail
}

```
You can just:

```
if (from.isAfter(to))  {
  fail
}
```

Reply:
Yeah you're right.